src/EOFeedVerifier.sol

```diff
@@ -414,19 +348,17 @@ contract EOFeedVerifier is IEOFeedVerifier, OwnableUpgradeable {
      * @param eventRoot event root the leaf should belong to
      * @return The leaf data field
      */
-    function _verifyLeaf(LeafInput calldata input, bytes32 eventRoot) internal view returns (bytes memory) {
+    function _verifyLeaf(LeafInput calldata input, bytes32 eventRoot) internal pure returns (bytes memory) {
         bytes32 leaf = keccak256(input.unhashedLeaf);
-        if (!leaf.checkMembership(input.leafIndex, eventRoot, input.proof)) {
+
+        if (!MerkleProof.verify(input.proof, eventRoot, leaf)) {
             revert InvalidProof();
         }
 
-        ( /* uint256 id */ , address sender, /* address receiver */, bytes memory data) =
-            abi.decode(input.unhashedLeaf, (uint256, address, address, bytes));
-        if (!_allowedSenders[sender]) {
-            revert SenderNotAllowed(sender);
-        }
+        // if (!leaf.checkMembership(input.leafIndex, eventRoot, input.proof)) {
+        // }
 
-        return data;
+        return input.unhashedLeaf;
```

Comment:
If we don't have `sender` field anymore inside the leaf, we can delete `_allowedSenders` mapping as it is not used